llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -207,6 +208,19 @@ attributeToExecModeMetadata(Module &M, const Attribute &Attr) {
                                             MDNode::get(Ctx, MD));
   }
 
+  // The sycl-single-task attribute currently only has an effect when targeting
+  // SPIR FPGAs, in which case it will generate a "max_global_work_dim" MD node
+  // with a 0 value, similar to applying [[intel::max_global_work_dim(0)]] to
+  // a SYCL single_target kernel.
+  if (AttrKindStr == "sycl-single-task" &&
+      Triple(M.getTargetTriple()).getSubArch() == Triple::SPIRSubArch_fpga) {
+    IntegerType *Ty = Type::getInt32Ty(Ctx);
+    Metadata *MDVal = ConstantAsMetadata::get(Constant::getNullValue(Ty));
+    SmallVector<Metadata *, 1> MD{MDVal};
+    return std::pair<std::string, MDNode *>("max_global_work_dim",
+                                            MDNode::get(Ctx, MD));
+  }
+
```

Comment:
What will happen when we apply [[intel::max_global_work_dim(1)]] to a SYCL single_target kernel. max_global_work_dim takes 0 and 1 value

Reply:
Like with other potentially conflicting Clang attributes and properties, the attribute takes precedence. That said, from what I am told it is not valid to use anything other than 0 for single_task kernels, so either we have invalid code or the Clang attribute has the same value as "sycl-single-task" attempts to add.

Reply:
Thanks @steffenlarsen for the explanation. Could you add test cases for value 1 ?